.travis.yml

```diff
@@ -14,249 +14,262 @@
 
 script:
 ####
 #  Install FreeCAD and run unit tests.  Test failures will fail the build
 ##
-  - sudo make -j2 install
-  - ${INSTALLED_APP_PATH} --run-test 0
-  - ${INSTALLED_APP_PATH} --log-file /tmp/FreeCAD_installed.log &
-  - sleep 10 && pkill FreeCAD
-  - cat /tmp/FreeCAD_installed.log
-  - grep --file=../.log_errors /tmp/FreeCAD_installed.log ; [ $? == 1 ] && echo "No errors from .log_errors file found in the log after start from /usr/local/bin" || ( echo "Error from .log_errors found!" && false )
+- sudo make -j2 install
+- ${INSTALLED_APP_PATH} --run-test 0
+- ${INSTALLED_APP_PATH} --log-file /tmp/FreeCAD_installed.log &
+- sleep 10 && pkill FreeCAD
+- cat /tmp/FreeCAD_installed.log
+- grep --file=../.log_errors /tmp/FreeCAD_installed.log ; [ $? == 1 ] && echo "No errors from .log_errors file found in the log after start from /usr/local/bin" || ( echo "Error from .log_errors found!" && false )
 
-after_success:
-####
-#  Package and deploy the build to GitHub.  This will only run for builds on
-#  master (i.e. pull requests are only built and tested but not deployed).
-#
-#  GH_TOKEN must be set in order to deploy releases to GitHub
-##
-- |
-  if [ "${TRAVIS_OS_NAME}" == "osx" -a "${TRAVIS_PULL_REQUEST}" == "false" -a "${DEPLOY}" == "1" ]; then
-     brew ls --versions jq || brew install jq
-     npm install -g appdmg
-     export VSN=$(python ${TRAVIS_BUILD_DIR}/src/Tools/ArchiveNameFromVersionHeader.py ${TRAVIS_BUILD_DIR}/build/src/Build/Version.h)
-     export DEPLOYMENT_ARCHIVE=${VSN}-${QT}.dmg
-     appdmg ${TRAVIS_BUILD_DIR}/src/MacAppBundle/DiskImage/layout.json "${DEPLOYMENT_ARCHIVE}"
-     deployContext=$(create_helper_context repo=${TRAVIS_REPO_SLUG} auth_token=${GH_TOKEN} release=${DEPLOY_RELEASE})
-     gitHub_deploy_asset_to_release_named $deployContext ${DEPLOYMENT_ARCHIVE}
-     gitHub_prune_assets_for_release_named $deployContext "-${QT}" 1
-  fi
+# Inject osx build into matrix - needed to specify image/dist
+jobs:
+    fast_finish: true
+    include:
+      - <<: *linux_base
+        <<: *clang_compiler
+        env: 
+          - CMAKE_OPTS="-DBUILD_FEM_NETGEN=ON"
+          - PYTHON_MAJOR_VERSION=2
+
+      - <<: *linux_base
+        <<: *gcc_compiler
+        env:
+          - CMAKE_OPTS="-DBUILD_FEM_NETGEN=ON"
+          - PYTHON_MAJOR_VERSION=2
+
+      - <<: *linux_base
+        <<: *gcc_compiler
+        env:
+          - CMAKE_OPTS="-DBUILD_FEM_NETGEN=ON"
+          - PYTHON_MAJOR_VERSION=3
+
+      - <<: *osx_base
+        <<: *clang_compiler
+        env:
+          - CMAKE_OPTS="-DBUILD_FEM_NETGEN=ON" QT=Qt5
+          - PYTHON_MAJOR_VERSION=2
+
+      - <<: *osx_base
+        <<: *clang_compiler
+        env:
+          - CMAKE_OPTS="-DBUILD_FEM_NETGEN=ON" QT=Qt4
+          - PYTHON_MAJOR_VERSION=2
+
+      - stage: Deploy dmg to GitHub Releases
+        # Prevent deploy job from appearing in tests unless commit is tagged
+        if: tag IS present
```

Comment:
See my comment in the main thread above about how this will change the current behavior.  Not sure why my first comment didn't grab the code snipped.

Reply:
Okay, I see, we could do two jobs with different constraints or change this to be combined constraint. Which one is preferable?

Reply:
I'll leave the decision of two jobs vs. one up to you in the context of your overall refactoring (i.e. whichever you think will work the best). We just want the end result to be that both the Qt5 & Qt4 builds are deployed here - https://github.com/FreeCAD/FreeCAD/releases/tag/0.17_pre